### PR TITLE
DM-23308: Add CMake support

### PR DESCRIPTION
**Pull request by fritzm**
Adds ability to build package with CMake in addition to eupspkg (allows to be included as submodule in other CMake projects, e.g. Qserv.)

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
+add_library(sphgeom SHARED)
+
+target_compile_features(sphgeom PRIVATE 
+    cxx_std_14
```
[the remark this answers is not in the document]
Comment (fritzm) [reply src/CMakeLists.txt]: Rolled to 17

**Comment by fritzm** [comment]
@timj GHA add to do cmake build and run C++ tests -- please have a look?

File: .github/workflows/cmake.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+      - name: CMake configure
+        run: cmake -B ${{github.workspace}}/build
+
+      - name: CMake build
+        run: cmake --build ${{github.workspace}}/build --parallel `nproc`
+
+      - name: CMake test
+        working-directory: ${{github.workspace}}/build
+        run: ctest  --parallel `nproc`
```
Comment (timj) [inline .github/workflows/cmake.yaml:31]: This test doesn't seem to do anything to test the python code. You are going to have to install pytest as I do in the other action and then run the tests to make sure it really does work from python.

Comment (fritzm) [reply .github/workflows/cmake.yaml]: @timj Added python tests to the cmake GHA; please have another look?

**Review by timj — APPROVED** [review]
Thanks for adding the python tests -- this makes it less likely someone will break cmake. If you also install `pytest-xdist` you can add `-n 2` to the pytest to make it run in parallel so be a little faster.

**Comment by fritzm** [comment]
Thanks @timj -- at less than 3s for the Python tests compared to the C++ build/link times, I'm inclined to not worry on it at this point. :-)

Let me know when the version tool lands, and I'll happily help out with the retrofitting of that into the cmake here and over in `log`.

**Pull request merged**